scripts/python/phdf.py

```diff
@@ -109,26 +109,24 @@ def __init__(self, filename):
             self.TotalCells = self.NumBlocks * self.CellsPerBlock
 
             # Read in coordinates
-            tmpx = f['/Locations/x'][:,:]
-            self.x = np.zeros((self.NumBlocks,self.MeshBlockSize[0]))
-            for bId in range(self.NumBlocks):
-                for cId in range(self.MeshBlockSize[0]):
-                    self.x[bId,cId] = 0.5*(tmpx[bId,cId]+tmpx[bId,cId+1])
-            self.xf = tmpx
-
-            tmpy = f['/Locations/y'][:,:]
-            self.y = np.zeros((self.NumBlocks,self.MeshBlockSize[1]))
-            for bId in range(self.NumBlocks):
-                for cId in range(self.MeshBlockSize[1]):
-                    self.y[bId,cId] = 0.5*(tmpy[bId,cId]+tmpy[bId,cId+1])
-            self.yf = tmpy
-
-            tmpz = f['/Locations/z'][:,:]
-            self.z = np.zeros((self.NumBlocks,self.MeshBlockSize[2]))
-            for bId in range(self.NumBlocks):
-                for cId in range(self.MeshBlockSize[2]):
-                    self.z[bId,cId] = 0.5*(tmpz[bId,cId]+tmpz[bId,cId+1])
-            self.zf = tmpz
+            def load_coord(coord_i):
+                coord_name = ["x","y","z"][coord_i]
+
+                tmp = f['/Locations/' + coord_name][:,:]
+                vol_loc = '/VolumeLocations/' + coord_name
+                if vol_loc in f:
+                    coord = f[vol_loc][:,:]
+                else:
+                    coord = np.zeros((self.NumBlocks,self.MeshBlockSize[coord_i]))
+                    for bId in range(self.NumBlocks):
+                        for cId in range(self.MeshBlockSize[coord_i]):
+                            coord[bId,cId] = 0.5*(tmp[bId,cId]+tmp[bId,cId+1])
```

Comment:
Wouldn't sth like
```suggestion
                    coord = 0.5*(tmp[:,:-1] + tmp[:,1:])
```
also work here as both for loops cover the entire range?

Reply:
Also, may be worth adding a warning here (printed) that this assume uniform Cartesian grids so that we don't miss it when adding other coordinate systems.